packages/fakta-ny-inntekt/src/components/tilkommetAktivitet/TilkommetAktivitetAccordion.tsx

```diff
@@ -0,0 +1,175 @@
+import dayjs from 'dayjs';
+
+import { TextAreaField } from '@navikt/ft-form-hooks';
+import { required } from '@navikt/ft-form-validators';
+import { AssessedBy } from '@navikt/ft-plattform-komponenter';
+import { VerticalSpacer } from '@navikt/ft-ui-komponenter';
+import { DDMMYYYY_DATE_FORMAT, ISO_DATE_FORMAT, TIDENES_ENDE } from '@navikt/ft-utils';
+
+import type {
+  TilkommetAktivitetFormValues,
+  TilkommetAktivitetValues,
+} from '../../types/FordelBeregningsgrunnlagPanelValues';
+import { FaktaFordelBeregningAvklaringsbehovCode } from '../../types/interface/FaktaFordelBeregningAvklaringsbehovCode.js';
+import { SubmitButton } from '../felles/SubmitButton';
+import { VurdertIForrigeBehandlingIcon } from '../felles/VurdertIForrigeBehandlingIcon';
+import { TidligereVurderteAktiviteterPanel } from './TidligereVurderteAktiviteterPanel';
+import { TilkommetAktivitetField } from './TilkommetAktivitetField';
+import { erVurdertTidligere, slaaSammenPerioder } from './TilkommetAktivitetUtils';
+
+import type { ArbeidsgiverOpplysningerPerId } from '../../types/ArbeidsgiverOpplysninger.js';
+import type { Beregningsgrunnlag } from '../../types/Beregningsgrunnlag.js';
+import type { VurderInntektsforholdPeriode } from '../../types/BeregningsgrunnlagFordeling.js';
+import styles from './tilkommetAktivitetAccordion.module.css';
+
+const formatDate = (date: string): string => (date ? dayjs(date, ISO_DATE_FORMAT).format(DDMMYYYY_DATE_FORMAT) : '-');
+
+const renderDateHeading = (fom: string, tom: string | undefined): ReactElement => {
+  if (!tom || tom === TIDENES_ENDE) {
+    return <Label size="medium">{`${formatDate(fom)} - `}</Label>;
+  }
+  return <Label size="medium">{`${formatDate(fom)} - ${formatDate(tom)}`}</Label>;
+};
+
+type Props = {
+  beregningsgrunnlag: Beregningsgrunnlag;
+  arbeidsgiverOpplysningerPerId: ArbeidsgiverOpplysningerPerId;
+  formName: string;
+  formFieldIndex: number;
+  readOnly: boolean;
+  submittable: boolean;
+  erAksjonspunktÅpent: boolean;
+  fields: TilkommetAktivitetValues[];
+};
+
+export const TilkommetAktivitetAccordion = ({
+  beregningsgrunnlag,
+  arbeidsgiverOpplysningerPerId,
+  formName,
+  formFieldIndex,
+  readOnly,
+  submittable,
+  erAksjonspunktÅpent,
+  fields,
+}: Props) => {
+  const [sammenslåttePerioder, setSammenslåttePerioder] = useState<VurderInntektsforholdPeriode[]>([]);
+  const [openPanels, setOpenPanels] = useState<string[]>([]);
+  const [alleFomDatoer, setAlleFomDatoer] = useState<string[]>([]);
+  const formMethods = useFormContext<TilkommetAktivitetFormValues>();
+
+  useEffect(() => {
+    const vurderInntektsforholdPerioder =
+      beregningsgrunnlag?.faktaOmFordeling?.vurderNyttInntektsforholdDto?.vurderInntektsforholdPerioder;
+    if (vurderInntektsforholdPerioder) {
+      const perioder = slaaSammenPerioder(vurderInntektsforholdPerioder, beregningsgrunnlag.forlengelseperioder);
+      setSammenslåttePerioder(perioder);
```

Comment:
Burde vi sjekke om sammenslåttePerioder og perioder har endret seg før vi setter den på nytt? Det kan forårsake unødvendige re-renders, eller blir det ikke relevant her?

```
if (JSON.stringify(perioder) !== JSON.stringify(sammenslåttePerioder)) {
    setSammenslåttePerioder(perioder);
}
```